src/models/Room.ts

```diff
@@ -0,0 +1,135 @@
+import { PowerLevelsEventContent } from "./events/PowerLevelsEvent";
+
+export enum RoomPreset {
```

Comment:
I've introduced enums here so that I could easily document what each of them do, but I realize that it's a breaking change in that existing code won't compile with enums. People often have opinions on enums, so I'd like yours.

Reply:
for this sort of stuff I'd prefer a `type` rather than enum, for reasons relating to enums being incompatible and weird to use. Modern IDEs should be capable of tab completing `type`s these days.